test/spec/services/hpaServiceSpec.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+
+describe('HPAService', function() {
+  var HPAService;
+  var $rootScope;
+  var $q;
+  var $window;
+
+  var metricsAreAvailable;
+
+  beforeEach(module('openshiftConsole', function($provide) {
+    metricsAreAvailable = true;
+    // MetricsService needs to be pass-through
+    var metricsServiceMock = {
+      isAvailable: function() {
+        return $q.when(metricsAreAvailable);
+      }
+    };
+    $provide.factory('MetricsService', function() {
+      return metricsServiceMock;
+    });
+  }));
+
+
+  beforeEach(inject(function(_$q_, _$rootScope_, _$window_, _HPAService_) {
+    $rootScope = _$rootScope_;
+    $q = _$q_;
+    $window = _$window_;
+    HPAService = _HPAService_;
+  }));
+
+  // Default fixtures.
+  // These are put together in such a way that they should not trigger any
+  // of the HPAService.getHPAWarnings().  Each test below should swap out the
+  // resource(s) fixture that will trigger the specific warning to test.
+  var defaultScaleTarget = {
+    kind: '',
+    spec: {
+      template: {
+        spec: {
+          containers: [
+            {
+              name: 'bob',
+              image: '',
+              env: [],
+              resources: {
+                requests: {
+                  memory: '32Mi',
+                  cpu: '250m'
+                },
+                limits: {
+                  memory: '64Mi',
+                  cpu: '250m'
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  };
+  var defaultHpaResources = [{}];
+  var defaultLimitRanges = [{
+    spec: {
+      limits: [{
+        type: '',
+        min: '',
+        max: '',
+        default: ''
+      }]
+    }
+  }];
+  var defaultProject = {
+    metadata: {
+      name: ''
+    }
+  };
+
+
+
+  describe('#addCPURequestWarning', function() {
+
+    describe('When there are no HPA resources', function() {
+      it('should return an empty set of messages (no errors)', function() {
+        var noHPA = [];
+        HPAService
+          .getHPAWarnings(defaultScaleTarget, noHPA, defaultLimitRanges, defaultProject)
+          .then(function(warnings) {
+            expect(_.size(warnings)).toEqual(0);
+          });
+        $rootScope.$digest();
+      });
+    });
+
+    describe('When metrics are not configured', function() {
+      it('should return a warning with reason MetricsNotAvailable', function() {
+        // flip the bool that controls the MetricsService mock
+        metricsAreAvailable = false;
+        HPAService
+          .getHPAWarnings(defaultScaleTarget, defaultHpaResources, defaultLimitRanges, defaultProject)
+          .then(function(warnings) {
+            expect(_.head(warnings).reason).toEqual('MetricsNotAvailable');
+          });
+        $rootScope.$digest();
+      });
+    });
+
+
+    // TODO: we should be testing more than just the happy paths
```

Comment:
Some of the tests here might be helpful:

https://github.com/openshift/origin-web-console/blob/master/test/spec/services/limitsRangesServiceSpec.js

Reply:
follow-on, or shall i flesh these out now?